In [1]:
!pip install transformers


In [2]:
from transformers import pipeline


In [3]:
!pip install transformers librosa torch


In [4]:
import librosa
import torch
from transformers import pipeline


In [ ]:
# Load pre-trained audio classification model
audio_classifier = pipeline("audio-classification")


In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

import warnings
warnings.filterwarnings("ignore")

In [6]:
import IPython.display as ipd
import librosa
import librosa.display

In [9]:
# Load pre-trained audio classification model
audio_classifier = pipeline("audio-classification")


No model was supplied, defaulted to superb/wav2vec2-base-superb-ks and revision 372e048 (https://huggingface.co/superb/wav2vec2-base-superb-ks).
Using a pipeline without specifying a model name and revision in production is not recommended.


config.json:   0%|          | 0.00/2.42k [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/378M [00:00<?, ?B/s]

preprocessor_config.json:   0%|          | 0.00/215 [00:00<?, ?B/s]

Device set to use cpu


In [16]:
from google.colab import files
uploaded = files.upload()
audio_file = list(uploaded.keys())[0]





Saving download.wav to download.wav


In [17]:
# Classify the audio/content/download (1).wav
results = audio_classifier("/content/download.wav")
print("Audio Classification Results:", results)


Audio Classification Results: [{'score': 0.7105070352554321, 'label': '_silence_'}, {'score': 0.28159648180007935, 'label': 'off'}, {'score': 0.004427511245012283, 'label': 'yes'}, {'score': 0.0018621365306898952, 'label': 'up'}, {'score': 0.001062785042449832, 'label': 'left'}]


Output Explanation:
[
    {'score': 0.7105, 'label': '_silence_'},
    {'score': 0.2816, 'label': 'off'},
    {'score': 0.0044, 'label': 'yes'},
    {'score': 0.0019, 'label': 'up'},
    {'score': 0.0011, 'label': 'left'}
]
Interpretation:
silence: The model has detected silence in the audio, with a 71.05% confidence.
off: The next most likely label is "off" (possibly indicating an off-sound, like a button turning off) with a 28.16% confidence.
yes: The model detected a slight probability (0.44%) of the label "yes", suggesting that there might have been a word "yes" in the audio.
up: Similarly, "up" is detected with a very low confidence (0.19%).
left: The model also detected "left", but with a very low probability (0.11%).
Why is this Happening?
The model is likely trained on a small set of predefined labels, and in this case, it has detected silence as the most prominent feature in your audio clip.
The audio clip may contain little to no speech or identifiable sounds that match the labels the model is trained on, leading to silence being the dominant classification.
Possible Actions to Improve Results:
Check the Audio Quality: Ensure that the audio clip is clear and not too noisy. If you're using a clip with a lot of background noise or silence, the model might classify it as silence.

Use Audio with Clearer Sound: Use a more recognizable sound, like a speech recording or specific sound effects (e.g., dog barking, music, etc.).

Try Different Audio: If you're working with a dataset that contains various sound labels, experiment with other sound files to see how the model performs.